terracotta/src/test/java/org/ehcache/ClusteredCacheManagerTest.java

```diff
@@ -0,0 +1,118 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache;
+
+import org.ehcache.config.BaseCacheConfiguration;
+import org.ehcache.config.BaseClusteredCacheSharedConfiguration;
+import org.ehcache.config.CacheRuntimeConfiguration;
+import org.ehcache.config.ClusteredCacheSharedConfiguration;
+import org.ehcache.config.ClusteredCacheSharedRuntimeConfiguration;
+import org.ehcache.config.EvictionPrioritizer;
+import org.ehcache.config.EvictionVeto;
+import org.ehcache.config.ResourcePoolsBuilder;
+import org.ehcache.config.TerracottaBaseCacheConfiguration;
+import org.ehcache.config.TerracottaCacheRuntimeConfiguration;
+import org.ehcache.config.TerracottaConfiguration;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.expiry.Expiry;
+import org.junit.Test;
+
+import java.net.URI;
+
+/**
+ * @author Alex Snaps
+ */
+public class ClusteredCacheManagerTest {
+
+  @Test
+  public void testNothingButAPI() {
+
+    final URI terracottaURI = sourceThat();
+    final TerracottaConfiguration tcConfig = new TerracottaConfiguration(terracottaURI);
+
+
+    // Pass TC Config in, and retrieve a ClusteredCacheManager to get access to live cycle stuff & server-side stuff (e.g. config)
+    final ClusteredCacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+        .with(tcConfig)
+        .build(true);
+
+    // UnclusteredCache
+    final Cache<String, Object> foo = cacheManager.createCache("foo",
+        new BaseCacheConfiguration<String, Object>(
+            String.class,
+            Object.class,
+            (EvictionVeto)null,        // Whatever this is
+            (EvictionPrioritizer)null, // Whatever this is
+            (Expiry)null,              // Whatever this is
+            this.getClass().getClassLoader(),
+            ResourcePoolsBuilder
+                .newResourcePoolsBuilder()
+                .heap(500, EntryUnit.ENTRIES)
+                .build())
+    );
+
+    // Clustered with all specified config stuff
+    final Cache<String, Object> baz = cacheManager.createCache("baz",
+        new TerracottaBaseCacheConfiguration<String, Object>(
+            new BaseClusteredCacheSharedConfiguration<String, Object>( // added this line
+                String.class,
+                Object.class,
+                (EvictionVeto)null,
+                (EvictionPrioritizer)null,
+                (Expiry)null,
+                2, MemoryUnit.TB, false, // 2 TB non-persistent, resourcePool aren't (yet?) "splittable"
+                false, // not transactional
+                BaseClusteredCacheSharedConfiguration.CacheLoaderWriter.NONE // no CacheLoaderWriter
```

Comment:
I don't know what you are asking?... In the dev meeting, we discussed that we _possibly_ wanted to know about certain "client" side constraints it then would enforce on all clients. That is a possible example of that. Though, in the case of WriteBehind, the server would need a WriterBehingQueue to enqueue matching operations for all mutative operations performed on a `Cache`, whether all clients consider it WriteBehind or not ... Now that client is actually creating it all, as such the requirement for the WriteBehindQueue _could_ be inferred. Again, that's just an example. I am _not_ saying the actual types are final. Far from it actually, it's the general approach that I would like first agreement on, then commitment sticking to it.